Fix for trivia not preserved after string interpolation conversion

AppAppWorks opened the pull request:
fix an issue in #1551

My bad, I should've tested the code action more thoroughly. The leading trivia of the source `SequenceExprSyntax` is not carried over to the new string literal, resulting in this:

```swift
<code block A>
"" + value + ""
<code block B>
```
->
```swift
<code block A>"\(value)"
<code block B>
```

This fix will preserve the leading trivia, i.e.
```swift
<code block A>
"\(value)"
<code block B>
```

ahoppen commented:
How was `// code block A` preserved before this change if the leading trivia of the code block weren’t preserved? Or did you mean for `// code block A` to be a statement?

Should we also retain the trailing trivia from `syntax` in the same way?

AppAppWorks commented:
Yeah,  `// code block A` was meant for a statement instead of a comment, I've changed that in the description above.

AFAIK, string literals don't have trailing trivia, and trailing trivia of any non-string-literal component will be converted and moved into the string interpolation, so the trailing trivia of `syntax` has been preserved in this way.

ahoppen commented:
What about `"a" + b + "c" / * some comment */`? In that case `/* some comment */` should be trailing trivia for the `"`.

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please test Windows

ahoppen commented on Tests/SourceKitLSPTests/CodeActionTests.swift:
```diff
@@ -1022,9 +1024,10 @@ final class CodeActionTests: XCTestCase {
+                  range: positions["0️⃣"]..<positions["5️⃣"],
                   newText: ###"""
-                    ##"[\##(key): \##(d) \##(value)]"##
+                    
```
Looks like this line has trailing whitespace. Could you run swift-format on your changes? https://github.com/swiftlang/sourcekit-lsp/blob/main/CONTRIBUTING.md#formatting

AppAppWorks replied:
Done, please test again.

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please test Windows
